sdk/storage/azure-storage-blobs/src/blob_lease_client.cpp

```diff
@@ -14,324 +15,325 @@ namespace Azure { namespace Storage { namespace Blobs {
     return Azure::Core::Uuid::CreateUuid().ToString();
   }
 
-  Azure::Response<Models::AcquireLeaseResult> BlobLeaseClient::Acquire(
-      std::chrono::seconds duration,
-      const AcquireLeaseOptions& options,
-      const Azure::Core::Context& context)
-  {
-    if (m_blobClient.HasValue())
-    {
-      _detail::BlobRestClient::Blob::AcquireBlobLeaseOptions protocolLayerOptions;
-      protocolLayerOptions.ProposedLeaseId = GetLeaseId();
-      protocolLayerOptions.LeaseDuration = duration;
-      protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
-      protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
-      protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
-      protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
-      protocolLayerOptions.IfTags = options.AccessConditions.TagConditions;
-
-      auto response = _detail::BlobRestClient::Blob::AcquireLease(
-          *(m_blobClient.Value().m_pipeline),
-          m_blobClient.Value().m_blobUrl,
-          protocolLayerOptions,
-          context);
-
-      Models::AcquireLeaseResult ret;
-      ret.ETag = std::move(response.Value.ETag);
-      ret.LastModified = std::move(response.Value.LastModified);
-      ret.LeaseId = std::move(response.Value.LeaseId);
-
-      return Azure::Response<Models::AcquireLeaseResult>(
-          std::move(ret), std::move(response.RawResponse));
-    }
-    else if (m_blobContainerClient.HasValue())
+#if defined(_MSC_VER)
+#pragma warning(push)
+// C4715: not all control paths return a value.
+// Either m_blobContainerClient or m_blobClient are expected.
+// Using ASSERT at the end to abort.
+#pragma warning(disable : 4715)
+#endif
+
```

Comment:
What's the benefit of this method over `std::abort`? Unreachable code is never supposed to be running anyway. What's the harm of using `std::abort`?

Reply:
We want to keep using abort(), but we basically want two things to be improved:
- Debug.  using assert give us a better debug experience.
- Single source of truth.  All abort calls can be controlled from the same place for the entire lib.